src/tyr/route_serializer_osrm.cc

```diff
@@ -1655,7 +1874,7 @@ std::string serialize(valhalla::Api& api) {
     route_summary(route, api, imperial, i);
 
     // Serialize route legs
-    route->emplace("legs", serialize_legs(api.directions().routes(i).legs(),
+    route->emplace("legs", serialize_legs(api.directions().routes(i).legs(), route_leg_summaries[i],
```

Comment:
Its splitting hairs I think.  Passing by const-ref is effectively like passing a pointer - very cheap.  The std::move() is likely a few instructions more... it would be an interesting thing to study and see which came out as more efficient.  In this usage/scenario I think I'll just leave as it.  An interesting insight to be sure though, thanks.

Reply:
Sorry that I didn't make it clear from the start :) Check it out https://github.com/valhalla/valhalla/pull/2874/files#r584849492

Reply:
So my idea is to move `route_leg_summaries[i]` so that we could move its elements inside `serialize_legs` function